contracts/Puzzle.sol

```diff
@@ -0,0 +1,69 @@
+pragma solidity >=0.4.22; 
+contract Puzzle {
+    string internal constant INSUFFICIENT_FUND_MESSAGE = "Insufficient Fund";
+    string internal constant RESTRICTED_MESSAGE = "Unauthorized Access";
+    uint amount;
+    mapping(address => uint) bestlevel;
+    address public manager;  // The adress of the owner of this contract
+    address payable[] public players;  // The players of current session
+
+    constructor() public {
+        manager = msg.sender;
+    }
+
+    /**
+     * @dev The player enters into the current game session by
+     *      paying at least 2 token.
+     */
+    function play() public payable {
+        require(msg.value == 2 ether, INSUFFICIENT_FUND_MESSAGE);
```

Comment:
>=?

Reply:
I actually did not want user to pay more than 2!

Reply:
will change it.